proxy/ProxyClientSession.h

```diff
@@ -101,6 +106,75 @@ class ProxyClientSession : public VConnection
   void do_api_callout(TSHttpHookID id);
 
   static int64_t next_connection_id();
+  virtual int get_transact_count() const = 0;
+
+  // Override if your session protocol allows this
+  virtual bool
+  is_transparent_passthrough_allowed()
+  {
+    return false;
+  }
+
+  // Override if your session protocol cares
+  virtual void
+  set_half_close_flag(bool flag)
+  {
+  }
+  virtual bool
+  get_half_close_flag()
```

Comment:
Yes, it should.  Fixing.  What compiler are you using?  It is giving better warnings.